web-bundle/src/main/java/com/graphhopper/http/GraphHopperManaged.java

```diff
@@ -319,21 +331,54 @@ public void setGtfsLinkMappings() {
+                    // Parse stable IDs and adjacent nodes for each edge from response
+                    List<PathDetail> responsePathEdgeIdDetails = response.getAll().get(0).getPathDetails().get("r5_edge_id");
+                    List<String> pathEdgeIds = responsePathEdgeIdDetails.stream()
                             .map(pathDetail -> (String) pathDetail.getValue())
                             .collect(Collectors.toList());
 
-                    // Remove first and last IDs, which are IDs of virtual GH edges
-                    if (pathStableEdgeIds.size() <= 2) {
-                        pathStableEdgeIds.clear();
+                    List<PathDetail> responsePathAdjNodeDetails = response.getAll().get(0).getPathDetails().get("adj_node");
+                    List<Integer> pathAdjNodes = responsePathAdjNodeDetails.stream()
+                            .map(pathDetail -> (Integer) pathDetail.getValue())
+                            .collect(Collectors.toList());
+
+                    EncodingManager encodingManager = graphHopper.getEncodingManager();
+                    StableIdEncodedValues stableIdEncodedValues = StableIdEncodedValues.fromEncodingManager(encodingManager);
+
+                    // Find indexes of edges in path that don't have stable edge IDs
+                    List<Integer> ghEdgeIdIndexesToConvert = Lists.newArrayList();
+                    for (int i = 0; i < pathEdgeIds.size(); i++) {
+                        // Check if edge has a GH edge ID by checking if it fits into Integer.MAX_VALUE;
+                        // IDs that are stable edge IDs will be >> Integer.MAX_VALUE
+                        try {
+                            // If the ID is a standard Java int, it's a gh edge ID
+                            Integer.parseInt(pathEdgeIds.get(i));
+                            ghEdgeIdIndexesToConvert.add(i);
+                        } catch (NumberFormatException e) {
+                            // ID must be > Integer.MAX_VALUE, so it's already a stable ID
+                        }
+                    }
+
+                    // Replace GH edge ID with stable edge ID for edges found above
+                    // todo: figure out how to do n=1 case. Can't seem to correctly link to proper GH edge in this case
+                    if (pathEdgeIds.size() == 1) {
+                        singleEdgeReturnedCount++;
+                        pathEdgeIds.clear();
                     } else {
-                        pathStableEdgeIds.remove(pathStableEdgeIds.size() - 1);
-                        pathStableEdgeIds.remove(0);
+                        for (int i : ghEdgeIdIndexesToConvert) {
+                            // todo: make sure this logic makes sense for adjacent node index for last virtual edge!
+                            int adjacentNodeIndex = (i == pathAdjNodes.size() - 1 && pathAdjNodes.size() != 1 ? pathAdjNodes.size() - 2 : i);
```

Comment:
I'll fix this, but @michaz I realized that because I added the check on L364-366 above, this can be simplified to 

`int adjacentNodeIndex = (i == pathAdjNodes.size() - 1 ? pathAdjNodes.size() - 2 : i);`

as you review this logic